wormhole-connect/src/hooks/useRoutesQuotesBulk.ts

```diff
@@ -0,0 +1,82 @@
+import { useState, useEffect, useMemo } from 'react';
+import { Chain, routes } from '@wormhole-foundation/sdk';
+
+import config from 'config';
+
+type RoutesQuotesBulkParams = {
+  sourceChain?: Chain;
+  sourceToken: string;
+  destChain?: Chain;
+  destToken: string;
+  amount: string;
+  nativeGas: number;
+};
+
+type QuoteResult = routes.QuoteResult<routes.Options>;
+
+type HookReturn = {
+  quotesMap: Record<string, QuoteResult | undefined>;
+  isFetching: boolean;
+};
+
+const useRoutesQuotesBulk = (
+  routes: string[],
+  params: RoutesQuotesBulkParams,
+): HookReturn => {
+  const [isFetching, setIsFetching] = useState(false);
+  const [quotes, setQuotes] = useState<QuoteResult[]>([]);
+
+  useEffect(() => {
+    let unmounted = false;
+    if (
+      !params.sourceChain ||
+      !params.sourceToken ||
+      !params.destChain ||
+      !params.destToken ||
+      !params.amount
+    ) {
+      return;
+    }
+
+    // Forcing TS to infer that fields are non-optional
+    const rParams = params as Required<RoutesQuotesBulkParams>;
+
+    setIsFetching(true);
+    config.routes
+      .computeMultipleQuotes(routes, rParams)
+      .then((quoteResults) => {
+        if (!unmounted) {
+          setQuotes(quoteResults);
+          setIsFetching(false);
```

Comment:
You need to set isFetching to false in an error case too. Like .finally()? Or even a try/catch to wrap the whole routes call.

Reply:
This is left intentionally, as `Promise.allSettled` can't reject per [spec](https://tc39.es/ecma262/#sec-promise.allsettled). 
Can add a comment or add a finally block just for readability.

Reply:
I see. Another thing is the promise callback comes here only once right? Shouldn't we call `setIsFetching(false)` outside the if statement?